### DISCUSSION TITLE
how to know the intersection line(s)

### DISCUSSION THREAD

**Discussion by ardeal** [shapely/shapely#2388] [Q&A]
Hi,

I would like to calculate the intersection of 2 polygon, and I also would like to know which line(s) is(are) the intersection line(s).
how to tell line(s)?

for example: 
for the following case, how to know the lines marked with red equal symbol?

![image](https://user-images.githubusercontent.com/6227348/101563955-2f6f4180-3a05-11eb-9572-d59ca1798ee0.png)

**Comment by mattijn** [comment]
I'm not sure you have read and applied the documentation regarding intersection?

https://shapely.readthedocs.io/en/stable/manual.html#object.intersection

```
from shapely import geometry
a = geometry.Polygon([(1, 2), (4, 2), (4, 4), (1, 4), (1, 2)])
b = geometry.Polygon([(0, 1), (1, 0), (3, 3), (1, 6), (0, 5), (0, 1)])
geometry.GeometryCollection([a, b])
```
<img width="88" alt="image" src="https://user-images.githubusercontent.com/5186265/101600496-aa335f00-39fb-11eb-8cd9-e52cdb906bb6.png">

```
a.intersection(b.boundary)
```
<img width="88" alt="image" src="https://user-images.githubusercontent.com/5186265/101600541-be775c00-39fb-11eb-9083-bfe142768352.png">

**Comment by ardeal** [comment]
Great！
Thank you, Mattijn!

One more questions:
suppose the left line of a polygon is parallel to the vertical line of b polygon. how to get the distance of the left up corner of a polygon to the vertical line of b polygon?
